tests/Input/LocalInputSourceTest.php

```diff
@@ -49,41 +59,99 @@ public function testPDFReadContents()
         $this->assertEquals("multipage.pdf", $contents[0]);
     }
 
-//    public function testPDFreconstructNoCut(){ // TODO when pdf handling lib is added
-//
-//    }
+    /**
+     * @dataProvider pageIndexesProvider
+     */
+    public function testPDFCutNPages(array $indexes)
+    {
+        $inputObj = new PathInput($this->fileTypesDir . "pdf/multipage.pdf");
+        $inputObj->processPDF(KEEP_ONLY, 2, $indexes);
+        try {
+            $basePdf = new FPDI();
+            $cutPdf = new FPDI();
+            $pageCountCutPdf = $cutPdf->setSourceFile(
+                $this->fileTypesDir . "pdf/multipage_cut-" . count($indexes) . ".pdf"
+            );
+            $pageCountBasePdf = $basePdf->setSourceFile($inputObj->fileObject->getFilename());
+            $basePdf->Close();
+            $cutPdf->Close();
+            $this->assertEquals(count($indexes), $inputObj->countDocPages());
+            $this->assertEquals($pageCountCutPdf, $pageCountBasePdf);
+
+            $basePdf = new FPDI();
+            $cutPdf = new FPDI();
+            for ($pageNumber = 0; $pageNumber < $pageCountBasePdf; $pageNumber++) {
+                $cutPdf->setSourceFile($this->fileTypesDir . "pdf/multipage_cut-" . count($indexes) . ".pdf");
+                $basePdf->setSourceFile($inputObj->fileObject->getFilename());
+                $cutPdf->AddPage();
+                $cutPdf->useTemplate($cutPdf->importPage($pageNumber + 1));
+                $basePdf->AddPage();
+                $basePdf->useTemplate($basePdf->importPage($pageNumber + 1));
+                // Note: comparing extracted page bytes content turns out to be reliable when using FPDF.
+                // This will be left here until a better solution is found within the limitations of licensing.
+                //                $this->assertEquals($cutPdf->Output('', 'S'), $basePdf->Output('', 'S'));
```

Comment:
not sure I understand your comment, shouldn't the assertion be executed, if it's reliable?

Reply:
typo, whoops